pkgs/development/python-modules/nose/default.nix

```diff
@@ -31,8 +32,9 @@ buildPythonPackage rec {
       --replace "from setuptools.command.build_py import Mixin2to3" "from distutils.util import Mixin2to3"
   '';
 
-  preBuild = lib.optionalString (isPy3k) ''
-    ${python.pythonOnBuildForHost}/bin/2to3 -wn nose functional_tests unit_tests
+  # 2to3 is removed from Python 3.13, so always use Python 3.12 2to3 for now.
+  preBuild = lib.optionalString isPy3k ''
+    ${python312.pythonOnBuildForHost}/bin/2to3 -wn nose functional_tests unit_tests
```

Comment:
I guess when 3.12 leaves Nixpkgs we’ll have to do one of:

* package it independently of the Python interpreter;
* run it once over the `nose` code and just vendor the results; or
* remove `nose`?

Reply:
Idealler by 2028 we've fixed the ecosystem enough, that nose is not relevant anymore.

Reply:
I didn’t realize that we kept Python releases for that many years. Agreed that it’s not worth thinking about in that case.